### ISSUE TITLE
Generic.Whitespace.ScopeIndent (exact) handles multi-line implements declaration incorrectly

### ISSUE THREAD

**Issue by tscni** [squizlabs/PHP_CodeSniffer#3376]
**Describe the bug**
Generic.Whitespace.ScopeIndent with exact=true handles implements declarations of classes, and doesn't do it well, but as with arrays indents. should probably disable the exact handling for that part?

**Code sample**
```php
<?php
class A implements
    B,
    C,
    D
{
}
```

**Custom ruleset**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<ruleset>
    <rule ref="Generic.WhiteSpace.ScopeIndent">
        <properties>
            <property name="exact" value="true" />
        </properties>
    </rule>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php`
3. See error message displayed
   ```
   FOUND 2 ERRORS AFFECTING 2 LINES
   -----------------------------------------------------------------------
    3 | ERROR | [x] Line indented incorrectly; expected 0 spaces, found 4
    4 | ERROR | [x] Line indented incorrectly; expected 0 spaces, found 4
   ```
2. Run `phpcbf test.php`
5. See formatting result
   ```php
   <?php
   class A implements
   B,
   C,
       D
   {
   }
   ```

**Expected behavior**
No error, no formatting change

**Versions (please complete the following information):**
 - PHP: 7.4
 - PHPCS: 3.6.0

**Possible fix**
```php
if ($tokens[$i]['code'] === T_CLASS) {
    $bracketStart = $phpcsFile->findNext(T_OPEN_CURLY_BRACKET, $i);
    if ($bracketStart !== false) {
        $disableExactEnd = max($disableExactEnd, $bracketStart - 1);
    }
}
```

**Comment by nelson6e65** [comment]
As a workaround, for the moment, you can disable `Generic.WhiteSpace.ScopeIndent` in those lines if you are extending PSR12:

```php
<?php 

// phpcs:disable Generic.WhiteSpace.ScopeIndent

class A implements
    B,
    C,
    D
{
    // phpcs:enable Generic.WhiteSpace.ScopeIndent
    
    // [...]
}

```
